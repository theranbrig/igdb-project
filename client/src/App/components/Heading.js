import React from 'react';
import { HeadingStyles } from '../styles/MainStyles';

const Heading = () => (
	<HeadingStyles>
		<h1>
			<i className="nes-icon nes-mario is-small" /> NintenDB{' '}
			<i className="nes-icon nes-kirby is-small" />
		</h1>
	</HeadingStyles>
);

export default Heading;
