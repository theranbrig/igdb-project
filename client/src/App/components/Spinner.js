import React from 'react';
import { PacmanLoader } from 'react-spinners';
import styled from 'styled-components';

const LoadingStyles = styled.div`
	.loading-screen {
		display: grid;
		grid-template-rows: 1fr;
		grid-template-columns: 1fr;
		min-height: 80vh;
		align-items: center;
		justify-items: center;
		width: 90%;
		margin-left: 5%;
		border: 3px solid #ffff00;
	}
`;

const Spinner = props => (
	<LoadingStyles>
		<div className="nes-container is-dark is-rounded loading-screen">
			<div className="sweet-loading">
				<PacmanLoader sizeUnit={'px'} size={50} color={'#ffff00'} loading={props.loading} />
			</div>
		</div>
	</LoadingStyles>
);

export default Spinner;
