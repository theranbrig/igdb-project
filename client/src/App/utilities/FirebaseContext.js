import React, { useState } from 'react';
import firebase from 'firebase';
import PropTypes from 'prop-types';

export const FirebaseContext = React.createContext();

const config = {
  apiKey: process.env.REACT_APP_API_KEY,
  authDomain: process.env.REACT_APP_AUTH_DOMAIN,
  databaseURL: process.env.REACT_APP_DATABASE_URL,
  projectId: process.env.REACT_APP_PROJECT_ID,
  storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
};

firebase.initializeApp(config);

const FirebaseProvider = props => {
  const [error, setError] = useState(null);
  const [authUser, setAuthUser] = useState(null);
  const [loading, setLoading] = useState(false);
  const [liked, setLiked] = useState(false);
  const [savedGames, setSavedGames] = useState([]);
  const [likedGame, setLikedGame] = useState(null);

  const db = firebase.firestore();

  const handleSignUp = async (email, password) => {
    setLoading(true);
    const user = await firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(setLoading(false))
      .catch(function(error) {
        setError(error.message);
        setLoading(false);
      });
    setAuthUser(user);
  };

  const handleSignIn = async (email, password) => {
    setLoading(true);
    const user = await firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(setLoading(false))
      .catch(function(error) {
        setError(error.message);
        setLoading(false);
      });
    setAuthUser(user);
  };

  const handleLogout = async () => {
    setLoading(true);
    await firebase.auth().signOut();
    setAuthUser(null);
    setLoading(false);
  };

  firebase.auth().onAuthStateChanged(function(user) {
    if (user) {
      setAuthUser(user);
    } else {
      console.log('No User Found');
    }
  });

  const checkGameSave = async (userId, gameId) => {
    setLiked(false);
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .where('gameId', '==', gameId)
      .get()
      .then(function(querySnapshot) {
        querySnapshot.forEach(function(doc) {
          if (doc.data()) {
            setLiked(true);
            setLikedGame(doc.data().likeId);
          }
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
  };

  const handleGameSave = async (userId, gameId, name, platform) => {
    await db
      .collection('votes')
      .add({
        userId,
        gameId,
        name,
        platform,
      })
      .then(async function(docRef) {
        await checkGameSave(userId, docRef.id);
      })
      .catch(function(error) {
        console.error('Error adding document: ', error);
      });
  };

  const handleGameUnsave = async (userId, gameId) => {
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .where('gameId', '==', gameId)
      .get()
      .then(function(querySnapshot) {
        querySnapshot.forEach(function(doc) {
          doc.ref.delete();
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
  };

  const checkSavedUserGames = async userId => {
    const tempSavedGames = [];
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .get()
      .then(querySnapshot => {
        querySnapshot.forEach(function(doc) {
          if (doc.data()) {
            tempSavedGames.push(doc.data());
          }
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
    setSavedGames(tempSavedGames);
  };

  return (
    <FirebaseContext.Provider
      value={{
        error,
        authUser,
        handleSignUp,
        handleSignIn,
        handleLogout,
        loading,
        handleGameSave,
        checkGameSave,
        checkSavedUserGames,
        liked,
        savedGames,
        setLoading,
        likedGame,
        handleGameUnsave,
        setLiked,
      }}
    >
      {props.children}
    </FirebaseContext.Provider>
  );
};

FirebaseProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default FirebaseProvider;
