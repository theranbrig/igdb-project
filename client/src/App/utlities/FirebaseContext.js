import React, { useState } from 'react';
import firebase from 'firebase';

export const FirebaseContext = React.createContext();

const config = {
	apiKey: process.env.REACT_APP_API_KEY,
	authDomain: process.env.REACT_APP_AUTH_DOMAIN,
	databaseURL: process.env.REACT_APP_DATABASE_URL,
	projectId: process.env.REACT_APP_PROJECT_ID,
	storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
	messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID
};

firebase.initializeApp(config);

const FirebaseProvider = props => {
	const [error, setError] = useState(null);
	// Auth User = authUser.uid /
	const [authUser, setAuthUser] = useState(null);
	const [loading, setLoading] = useState(false);
	const [liked, setLiked] = useState(false);
	const [savedGames, setSavedGames] = useState([]);

	const db = firebase.firestore();

	const handleSignUp = async (email, password) => {
		setLoading(true);
		const user = await firebase
			.auth()
			.createUserWithEmailAndPassword(email, password)
			.then(() => db.collection('users').add({ email }))
			.then(docRef => {
				db.collection('users')
					.doc(docRef.id)
					.update({ userId: docRef.id });
				setLoading(false);
			})
			.catch(function(error) {
				setError(error.message);
				setLoading(false);
			});
		setAuthUser(user);
	};

	const handleSignIn = async (email, password) => {
		setLoading(true);
		const user = await firebase
			.auth()
			.signInWithEmailAndPassword(email, password)
			.then(setLoading(false))
			.catch(function(error) {
				setError(error.message);
				setLoading(false);
			});
		setAuthUser(user);
	};

	const handleLogout = async () => {
		setLoading(true);
		await firebase.auth().signOut();
		setAuthUser(null);
		setLoading(false);
	};

	firebase.auth().onAuthStateChanged(function(user) {
		if (user) {
			setAuthUser(user);
		} else {
			console.log('No User Found');
		}
	});

	const handleGameSave = async (userId, gameId, name, platform) => {
		const like = await db
			.collection('votes')
			.add({
				userId,
				gameId,
				name,
				platform
			})
			.then(function(docRef) {
				console.log('Document written with ID: ', docRef.id);
			})
			.catch(function(error) {
				console.error('Error adding document: ', error);
			});
	};

	const checkGameSave = async (userId, gameId) => {
		setLiked(false);
		const saveCheck = await db
			.collection('votes')
			.where('userId', '==', userId)
			.where('gameId', '==', gameId)
			.get()
			.then(function(querySnapshot) {
				querySnapshot.forEach(function(doc) {
					if (doc.data()) {
						setLiked(true);
					}
				});
			})
			.catch(function(error) {
				console.log('Error getting documents: ', error);
			});
	};

	const checkSavedUserGames = async userId => {
		const tempSavedGames = [];
		const response = await db
			.collection('votes')
			.where('userId', '==', userId)
			.get()
			.then(querySnapshot => {
				querySnapshot.forEach(function(doc) {
					if (doc.data()) {
						tempSavedGames.push(doc.data());
					}
				});
			})
			.catch(function(error) {
				console.log('Error getting documents: ', error);
			});
		setSavedGames(tempSavedGames);
	};

	return (
		<FirebaseContext.Provider
			value={{
				error,
				authUser,
				handleSignUp,
				handleSignIn,
				handleLogout,
				loading,
				handleGameSave,
				checkGameSave,
				checkSavedUserGames,
				liked,
				savedGames
			}}>
			{props.children}
		</FirebaseContext.Provider>
	);
};

export default FirebaseProvider;
