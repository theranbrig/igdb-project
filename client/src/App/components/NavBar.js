import React, { useContext } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { FirebaseContext } from '../utlities/FirebaseContext';

const NavStyles = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  height: 100px;
  .nav-logo,
  .nav-links {
    display: inline-flex;
    align-self: center;
    p {
      font-size: 2rem;
      margin-left: 10px;
    }
    a {
      display: inherit;
      color: black;
      font-family: 'Press Start 2p';
      &:hover {
        text-decoration: none;
        text-shadow: 1px 1px 10px gray;
      }
    }
  }
  .nav-logo {
    margin-left: 10px;
  }
  .nav-links button {
    margin: 10px;
    font-family: 'Press Start 2p';
  }
`;

const NavBar = () => {
  const { authUser } = useContext(FirebaseContext);
  return (
    <NavStyles>
      <div className="nav-logo">
        <Link to="/">
          <i className="nes-logo" />
          <p>NintenDB</p>
        </Link>
      </div>
      <div className="nav-links">
        <Link to={{ pathname: '/random', query: { platform: 18 } }}>
          <button className="nes-btn is-success">Random Game</button>
        </Link>
        {authUser ? (
          <Link to="/mypage">
            <button className="nes-btn is-primary">My Saved Games</button>
          </Link>
        ) : (
          <>
            <Link to="/signup">
              <button className="nes-btn is-primary">Sign Up</button>
            </Link>
            <Link to="/login">
              <button className="nes-btn is-primary">Login</button>
            </Link>
          </>
        )}
      </div>
    </NavStyles>
  );
};

export default NavBar;
