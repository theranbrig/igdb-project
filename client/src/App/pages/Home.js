import styled from 'styled-components';
import React from 'react';
import Form from '../components/Form';
import Heading from '../components/Heading';
import NavBar from '../components/NavBar';

const MainStyles = styled.div`
	.App {
		margin: 20px;
		min-height: 80vh;
		display: grid;
		grid-template-rows: 1fr;
		grid-template-columns: 1fr;
		min-height: 80vh;
		align-items: center;
		justify-items: center;
		margin: 10vh auto;
		width: 90%;
		.title {
			font-family: 'Press Start 2p';
		}
	}
	h1,
	i {
		padding: 15px 0 0;
		font-family: 'Press Start 2p';
		font-size: 5rem;
		text-align: center;
	}
	p {
		font-family: 'Press Start 2p';
	}
	.loading-screen {
		display: grid;
		grid-template-rows: 1fr;
		grid-template-columns: 1fr;
		min-height: 80vh;
		align-items: center;
		justify-items: center;
		margin: 10vh auto;
	}
	.icon-list {
		margin: 30px;
		i {
			margin: 20px !important;
		}import Heading from './Heading';
import NavBar from '../components/NavBar';

	}
`;

const Home = () => (
  <MainStyles>
    <NavBar />
    <div className="App nes-container is-dark is-rounded">
      <Heading />
      <Form />
      <section className="icon-list">
        <i className="nes-ash" />
        <i className="nes-pokeball" />
        <i className="nes-bulbasaur" />
        <i className="nes-charmander" />
        <i className="nes-squirtle" />
      </section>
    </div>
  </MainStyles>
);

export default Home;
