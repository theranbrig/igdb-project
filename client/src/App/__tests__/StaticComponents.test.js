import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Heading from '../components/Heading';
import Home from '../pages/Home';
import { FirebaseContext } from '../utilities/FirebaseContext';
import NavBar from '../components/NavBar';
import Spinner from '../components/Spinner';

const authUser = { email: 'fake@fake.com' };
const nonAuthUser = null;

it('renders the navbar without user', () => {
	const element = TestRenderer.create(
		<Router>
			<FirebaseContext.Provider value={{ authUser: nonAuthUser }}>
				<NavBar />
			</FirebaseContext.Provider>
		</Router>
	).toJSON();
	expect(element).toMatchSnapshot();
});

it('renders the navbar with a user', () => {
	const element = TestRenderer.create(
		<Router>
			<FirebaseContext.Provider value={{ authUser }}>
				<NavBar />
			</FirebaseContext.Provider>
		</Router>
	).toJSON();
	expect(element).toMatchSnapshot();
});

it('renders heading correctly', () => {
	const heading = TestRenderer.create(<Heading />).toJSON();
	expect(heading).toMatchSnapshot();
});

it('renders the home page correctly', () => {
	const element = TestRenderer.create(
		<Router>
			<FirebaseContext.Provider value={{ authUser }}>
				<Home />
			</FirebaseContext.Provider>
		</Router>
	).toJSON();
	expect(element).toMatchSnapshot();
});

it('renders the home page correctly', () => {
	const element = TestRenderer.create(
		<Router>
			<FirebaseContext.Provider value={{ loading: true }}>
				<Spinner loading={true} />
			</FirebaseContext.Provider>
		</Router>
	).toJSON();
	expect(element).toMatchSnapshot();
});
