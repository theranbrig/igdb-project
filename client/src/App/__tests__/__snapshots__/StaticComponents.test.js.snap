// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heading correctly 1`] = `
<div
  className="sc-bxivhb bbnBSQ"
>
  <h1>
    <i
      className="nes-icon nes-mario is-small"
    />
     NintenDB 
    <i
      className="nes-icon nes-kirby is-small"
    />
  </h1>
</div>
`;

exports[`renders the home page correctly 1`] = `
<div
  className="sc-bdVaJa dpQpIH"
>
  <div
    className="sc-htpNat jdiiWr"
  >
    <div
      className="nav-logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="nes-logo"
        />
        <p>
          NintenDB
        </p>
      </a>
    </div>
    <div
      className="nav-links"
    >
      <a
        href="/random"
        onClick={[Function]}
      >
        <button
          className="nes-btn is-success"
          type="button"
        >
          Random Game
        </button>
      </a>
      <a
        href="/mypage"
        onClick={[Function]}
      >
        <button
          className="nes-btn is-primary"
          type="button"
        >
          My Saved Games
        </button>
      </a>
    </div>
  </div>
  <div
    className="App nes-container is-dark is-rounded"
  >
    <div
      className="sc-bxivhb bbnBSQ"
    >
      <h1>
        <i
          className="nes-icon nes-mario is-small"
        />
         NintenDB 
        <i
          className="nes-icon nes-kirby is-small"
        />
      </h1>
    </div>
    <ul>
      <li>
        Click to get a random Nintendo game from your desired platform and learn more about it.
      </li>
      <li>
        Sign Up and Login to save your games for later viewing
      </li>
      <li>
        Click the heart icon to save your favorite games.
      </li>
    </ul>
    <div
      className="sc-EHOje bCUIMy"
    >
      <form>
        <label
          className="nes-text is-primary"
          htmlFor="platform"
        >
          Choose Your Platform Type
        </label>
        <div
          className="nes-select"
        >
          <select
            defaultValue=""
            id="platform"
            name="platform"
            onChange={[Function]}
            required={true}
          >
            <option
              value="18"
            >
              Nintendo
            </option>
            <option
              value="19"
            >
              Super Nintendo
            </option>
            <option
              value="4"
            >
              N64
            </option>
            <option
              value="21"
            >
              Game Cube
            </option>
          </select>
        </div>
        <a
          href="/random"
          onClick={[Function]}
        >
          <button
            className="nes-btn is-success"
            type="button"
          >
            Random Game
          </button>
        </a>
      </form>
    </div>
    <section
      className="icon-list"
    >
      <i
        className="nes-ash"
      />
      <i
        className="nes-pokeball"
      />
      <i
        className="nes-bulbasaur"
      />
      <i
        className="nes-charmander"
      />
      <i
        className="nes-squirtle"
      />
    </section>
  </div>
</div>
`;

exports[`renders the home page correctly 2`] = `
<div
  className="sc-bwzfXH kDAQSs"
>
  <div
    className="nes-container is-dark is-rounded loading-screen"
  >
    <div
      className="sweet-loading"
    >
      <div
        className="css-hbuhc5-wrapper-wrapper"
      >
        <div
          className="css-1e7sjhp-pacmanStyle"
        />
        <div
          className="css-306891-pacmanStyle"
        />
        <div
          className="css-ph48sd-ballStyle"
        />
        <div
          className="css-cnn7k-ballStyle"
        />
        <div
          className="css-1nzbglu-ballStyle"
        />
        <div
          className="css-1p1zyao-ballStyle"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the navbar with a user 1`] = `
<div
  className="sc-htpNat jdiiWr"
>
  <div
    className="nav-logo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <i
        className="nes-logo"
      />
      <p>
        NintenDB
      </p>
    </a>
  </div>
  <div
    className="nav-links"
  >
    <a
      href="/random"
      onClick={[Function]}
    >
      <button
        className="nes-btn is-success"
        type="button"
      >
        Random Game
      </button>
    </a>
    <a
      href="/mypage"
      onClick={[Function]}
    >
      <button
        className="nes-btn is-primary"
        type="button"
      >
        My Saved Games
      </button>
    </a>
  </div>
</div>
`;

exports[`renders the navbar without user 1`] = `
<div
  className="sc-htpNat jdiiWr"
>
  <div
    className="nav-logo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <i
        className="nes-logo"
      />
      <p>
        NintenDB
      </p>
    </a>
  </div>
  <div
    className="nav-links"
  >
    <a
      href="/random"
      onClick={[Function]}
    >
      <button
        className="nes-btn is-success"
        type="button"
      >
        Random Game
      </button>
    </a>
    <a
      href="/signup"
      onClick={[Function]}
    >
      <button
        className="nes-btn is-primary"
        type="button"
      >
        Sign Up
      </button>
    </a>
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="nes-btn is-primary"
        type="button"
      >
        Login
      </button>
    </a>
  </div>
</div>
`;
